### DISCUSSION TITLE
[ZK Stack] - Long bridging time for Abstract Testnet

### DISCUSSION THREAD

**Discussion by coffee-the-dev** [zkSync-Community-Hub/zksync-developers#721] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [X] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

N/A

### Operating System

- [ ] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We had a long period of L1 -> L2 bridging transactions not going through on abstract testnet - it looks like they all cleared about 
08:20 UTC but there was a period of at least 30 hours where bridge transactions were not clearing.

How can we troubleshoot bridge transactions and ensure timely delivery for L1 -> L2 bridging?

### Expected Behavior

Bridge transactions should clear in ~15 minutes

### Current Behavior

No bridge transactions cleared for at least 30 hours. Block 460163 contained all bridging transactions from many hours.
https://explorer.testnet.abs.xyz/block/460163

### Additional Details

Example 1:
Sepolia: https://sepolia.etherscan.io/tx/0x1f28037317110da956180feda09700456d256534da242a3d4192946d6f9d26c0
Abstract Testnet: https://explorer.testnet.abs.xyz/tx/0x419c0cd93e453034e7d31a4bfcbb63606d400037bd475ad0d5170f62f097dca9

Example 2: 
Sepolia (Oct 8 2024, 1:00 UTC): https://sepolia.etherscan.io/tx/0xe7cfea5ef1eb81c9cacb36ae7ab0395d9b5c825d1fed130f734c151136455d39
Abstract Testnet (Oct 9 2024, 8:20 UTC): https://explorer.testnet.abs.xyz/tx/0x9f163bc740eda2d97cf1facb9c06c6ceae096caea1d0c0fa97e5a06f2b5a2aee

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To troubleshoot and ensure timely delivery for L1 to L2 bridging transactions on the Abstract Testnet, you can follow these steps:

1. **Check Network Status**: Verify the current status of the Abstract Testnet and any ongoing maintenance or known issues that might affect transaction processing.

2. **Review Transaction Details**:
   - Ensure that the transaction on L1 (Sepolia) has been confirmed and has sufficient gas.
   - Check the corresponding transaction on L2 (Abstract Testnet) for any errors or unusual statuses.

3. **Monitor Block Production**:
   - Observe the block times and sizes on the Abstract Testnet to identify any anomalies or delays in block production that could affect bridging.

4. **Consult Logs and Metrics**:
   - If possible, review server logs and performance metrics for the bridging infrastructure to identify bottlenecks or errors during the affected period.

5. **Engage with Community or Support**:
   - If the issue persists, consider reaching out to the ZKsync community or support channels for further assistance and to report the issue.

For ongoing issues or if you need more detailed guidance, engaging directly with the support or development team responsible for the Abstract Testnet might provide more specific insights and solutions.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
Thanks for the question! Let me investigate the issue.

**Comment by dutterbutter** [comment, marked as answer]
@coffeexcoin It looks like this issue was a result of a previous deployment that has now been reverted and fixed. This was not an issue with the portal itself as the portal is only responsible for initiating and sending deposit transaction. 

Specifically the following [release](https://github.com/matter-labs/zksync-era/releases/tag/core-v24.28.0) was deployed that included these [changes](https://github.com/matter-labs/zksync-era/pull/2867/files#diff-53284e014bc22d44b835d8c785843d1a1b5ae6253768ae1cfe6e7ed79e170532) which seem to break on Abstract. These changes have since been reverted and fixed. 

Hopefully that provides some clarity! 😄

**Reply by coffee-the-dev** [reply]
Thanks - is there a way we can get better monitoring/alerting for this kind of scenario going forward?

**Reply by dutterbutter** [reply]
Let me discuss with the team what can be done here as it seems quite sensible to have observability for these situations if there isn't already!
